src/istgt_lu_ctl.c

```diff
@@ -608,6 +608,71 @@ istgt_uctl_cmd_snap(UCTL_Ptr uctl)
 	return (ret);
 }
 
+static int
+istgt_uctl_cmd_resize(UCTL_Ptr uctl)
+{
+	ISTGT_LU_Ptr lu = NULL;
+	ISTGT_LU_DISK *spec = NULL;
+	int rc = 0, ret = UCTL_CMD_ERR;
+	char *volname, *size_str, *arg;
+	uint64_t new_size, old_size;
+	bool r;
+	const char *delim = ARGS_DELIM;
+	arg = uctl->arg;
+
+	CHECK_ARG_AND_GOTO_ERROR;
+	volname = strsepq(&arg, delim);
+
+	CHECK_ARG_AND_GOTO_ERROR;
+	size_str = strsepq(&arg, delim);
+
+	if (size_str == NULL) {
+		istgt_uctl_snprintf(uctl, "LUN format error\n");
+		goto error_return;
+	} else if (*size_str == '-') {
+		istgt_uctl_snprintf(uctl, "invalid size\n");
+		goto error_return;
+	}
+	ISTGT_LOG("volume(%s) requested to resize %s\n", volname, size_str);
+
+	lu = istgt_lu_find_target_by_volname(uctl->istgt, volname);
+	if (lu == NULL) {
+		istgt_uctl_snprintf(uctl, "ERR target not found\n");
+		goto error_return;
+	}
+	spec = lu->lun[0].spec;
+	//old_size = lu->lun[0].u.storage.size;
+	old_size = spec->size;
+	new_size = istgt_lu_parse_size(size_str);
+	if (old_size >= new_size) {
+		ISTGT_LOG("volume(%s) size is already greater than or equal to requested resize\n", volname);
+		istgt_uctl_snprintf(uctl, "OK %s\n", uctl->cmd);
+		ret = UCTL_CMD_OK;
```

Comment:
here we are returning success in this case. How the pool_mgmt  knows that it should not go ahead and update the size? Does it  verifies that size is larger than only update?

Reply:
Good catch Sir I will return error